hydra-node/src/Hydra/Node.hs

```diff
@@ -142,12 +142,13 @@ stepHydraNode tracer node = do
       forM_ effs (processEffect node tracer)
     OnlyEffects effs ->
       forM_ effs (processEffect node tracer)
-  traceWith tracer (EndEvent party e)
+  traceWith tracer EndEvent{by = party, eventId}
  where
   decreaseTTL =
     \case
       -- XXX: this is smelly, handle wait re-enqueing differently
-      NetworkEvent ttl msg | ttl > 0 -> NetworkEvent (ttl - 1) msg
+      Queued{eventId, eventQueued = NetworkEvent ttl msg}
+        | ttl > 0 -> Queued{eventId, eventQueued = NetworkEvent (ttl - 1) msg}
```

Comment:
This makes it now very clear that we should do this TTL business on the `Queued` type and discard any events when it runs out of "lives". No need to change now, but maybe a follow-up PR? :)

Reply:
Agreed :)